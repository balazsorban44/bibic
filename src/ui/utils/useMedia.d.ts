declare function useMedia(queryString: string) : boolean
declare function useMedia(queryObject: {
  direction: "up" | "down",
  size: number
}) : boolean

export default useMedia