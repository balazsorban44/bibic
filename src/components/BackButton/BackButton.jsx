import React from "react"
import {withRouter, Link} from "react-router-dom"
import {useTranslation} from "react-i18next"

const BackButton = withRouter(props => {
  const [t] = useTranslation("common")
  const {location: {search}, history: {goBack}} = props
  const customReturn = new URLSearchParams(search).get("vissza")
  const title = t(customReturn || "homepage")
  return(
    customReturn ?
      <button
        className="back-to-home"
        onClick={goBack}
        title={title}
      >
        <Title {...{title}}/>
      </button> :
      <Link
        className="back-to-home"
        title={title}
        to="/"
      >
        <Title {...{title}}/>
      </Link>
  )
})

const Title = ({title}) => {
  const [t] = useTranslation("common")
  return (
    <>
      ‚Üê
      <span>
        {t("back-to")} {t(title)}
      </span>
    </>
  )
}

export default BackButton