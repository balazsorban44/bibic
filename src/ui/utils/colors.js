const rooms = {
  room1: "#724444",
  room2: "#5e9782",
  room3: "#b35d41",
  room4: "#844644",
  room5: "#405572",
  room6: "#5e7240"
}

const prices = {
  price1: "#392222",
  price2: "#0c262c",
  price3: "#0c2c22",
  price4: "#272727"
}
const colors = {
  accentColor: "#482c29",
  accentColor2: "#d3b098",
  accentColor3: "#38211e",
  accentColor4: "#594630",
  accentColor5: "#c77a7a",
  accentColorDark: "#1c1614",
  lightred: "#dd6565",
  light: "#fafafa",
  grey: "#c7c7c7",
  dark: "black",
  warning: "#c4960d",
  error: "#7f0606",
  success: "#30893f",
  rooms,
  prices
}

export default colors