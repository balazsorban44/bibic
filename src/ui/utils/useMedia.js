import {useEffect, useState} from "react"

const getMediaQuery = (direction, size) =>
  `(${direction === "up" ? "max-width" : "min-width"}: ${size}px)`

const useMedia = query => {

  let queryString = query
  if (typeof query !== "string") {
    const {direction, size} = query
    queryString = getMediaQuery(direction, size)
  }

  const [matches, setMatches] = useState(window.matchMedia(queryString).matches)

  useEffect(() => {
    const matcher = () => {
      const _matches = window.matchMedia(queryString).matches
      matches !== _matches && setMatches(_matches)
    }

    window.addEventListener("resize", matcher)
    return () => window.removeEventListener("resize", matcher)
  }, [matches, queryString])

  return matches
}

export default useMedia