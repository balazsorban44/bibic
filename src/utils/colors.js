export default {
  accent: {
    primary: "#482c29",
    primaryLight: "#594630",
    primaryDark: "#1c1614",
    primaryDarker: "#38211e",
    secondary: "#d3b098"
  },
  lightred1: "#c77a7a",
  lightred2: "#dd6565",
  light: "#fafafa",
  grey: "#c7c7c7",
  dark: "#000000",
  warning: "#c4960d",
  error: "#7f0606",
  success: "#30893f",
  rooms: [ "#724444", "#5e9782", "#b35d41", "#844644", "#405572", "#5e7240"],
  prices: [ "#392222", "#0c262c", "#0c2c22", "#272727"]
}