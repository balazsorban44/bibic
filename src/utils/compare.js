export const isEquivalent = (a, b) => {
  // Create arrays of property names
  const aProps = Object.getOwnPropertyNames(a)
  const bProps = Object.getOwnPropertyNames(b)

  // If number of properties is different, objects are not equivalent
  if (aProps.length !== bProps.length) return false

  for (let i = 0; i < aProps.length; i++) {
    const propName = aProps[i]
    // If values of same property are not equal, objects are not equivalent
    if (a[propName] !== b[propName]) return false
  }

  // If we made it this far, objects are considered equivalent
  return true
}