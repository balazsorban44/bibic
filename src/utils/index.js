if (process.env.NODE_ENV === "test")
  Object.fromEntries = o => o.reduce((acc, [k, v]) => {
    acc[k] = v
    return acc
  }, {})


/**
 * Filters an Object by keys
 * @param {*} obj Object to filter
 * @param {Array} filteredKeys the keys that should be preserved
 */
export function filterObject(obj, filteredKeys) {
  return Object.fromEntries(
    Object.entries(obj).filter(([key]) => filteredKeys.includes(key))
  )
}