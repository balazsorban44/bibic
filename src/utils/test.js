import {render} from "@testing-library/react"
import {MemoryRouter} from "react-router-dom"
import {I18nextProvider} from "react-i18next"
import i18n from "lib/i18n"
import Database from "db"

const wrapper = ({children}, {context={}, router={}}) =>
  <I18nextProvider i18n={i18n}>
    <MemoryRouter {...router}>
      <Database context={context}>
        {children}
      </Database>
    </MemoryRouter>
  </I18nextProvider>


const customRender = (ui, options={}) =>
  render(ui, {wrapper: props => wrapper(props, {router: options.router, context: options.context}), ...options})


export * from "@testing-library/react"

export {customRender as render}


