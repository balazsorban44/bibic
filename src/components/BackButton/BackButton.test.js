import BackButton from "./BackButton"
import {render} from "utils/test"

describe("BackButton", () => {

  it("default to homepage", () => {
    const {container} = render(<BackButton/>)
    expect(container).toMatchSnapshot()
    expect(container).toHaveTextContent(/back.*to.*home/i)
  })

  it("custom", () => {
    // Custom return i.e. back button will go to another page than homepage
    const custom = "custom"
    const {container} = render(<BackButton />, {router: {initialEntries: [`something/?vissza=${custom}`]}})
    expect(container).toHaveTextContent(custom)
  })
})
