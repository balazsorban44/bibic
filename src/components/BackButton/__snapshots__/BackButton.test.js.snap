// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton default to homepage 1`] = `
<div>
  <a
    class="back-to-home"
    href="/"
    title="home"
  >
    ‚Üê
    <span>
      Back to
       
      home
    </span>
  </a>
   
</div>
`;
